### ISSUE TITLE
Bug: ignore `<a>` elements without `href` in `require-link-title`

### ISSUE THREAD

**Issue by electrohyun** [eslint-markdown/eslint-markdown#719]
### What rule do you want to change?

`require-link-title`

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
<a id="section"></a>
```

### What does the rule currently do for this code?

The rule reports that the `<a>` element is missing a title attribute.

[reproduce](https://stackblitz.com/edit/stackblitz-starters-6yjgiguu?file=repro.md)

(run `npx eslint repro.md`)

### What will the rule do after it's changed?

The rule will ignore `<a>` elements without an `href` attribute.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] No - I did not use AI to generate this issue report.
- [ ] Yes - minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes - substantive assistance (AI-generated non-trivial portions of code, tests, or documentation).

### Additional comments

I think `<a>` elements without an `href` may be outside the scope of this rule.

The [HTML Standard’s definition of the `<a>` element](https://html.spec.whatwg.org/dev/text-level-semantics.html#the-a-element) explains that an `<a>` element without an `href` represents a placeholder. 

> If the a element has no href attribute, then the element represents a placeholder for where a link might otherwise have been placed, if it had been relevant, consisting of just the element's contents.

The section on [links created by `<a>` elements](https://html.spec.whatwg.org/dev/links.html#links-created-by-a-and-area-elements) states that `href` is not required and that elements without it do not create hyperlinks. In addition, all HTML examples in the rule documentation include an `href`.

> The href attribute on a and area elements is not required; when those elements do not have href attributes they do not create hyperlinks.

For these reasons, I think the rule should ignore `<a>` elements without an `href`. Disabling the rule would also stop checking actual hyperlinks. I would appreciate your thoughts on whether this behavior is intended.

**Comment by lumirlumir** [comment]
Thanks for the issue! I hadn’t thought about this before, but since the `require-link-title` rule is about links, an `a` element without an `href` isn’t actually an HTML link and doesn’t behave like one. Therefore, this behavior makes sense to me.

That said, I have one question: what should the behavior be for `require-image-title`? Could an `img` element without a `src` also be treated as a placeholder rather than an image? Since `require-link-title` and `require-image-title` are sibling rules, it would be nice to handle this case consistently there as well.

It'd be helpful if you can give me some examples about `img` tags. 🙂

**Comment by electrohyun** [comment]
I generally prefer sibling rules to be consistent, as I think that makes them "prettier." However, I think we should follow the HTML Standard here.

According to [the definition of the `<img>` element](https://html.spec.whatwg.org/dev/embedded-content.html#the-img-element) , at least one of the `src` and `srcset` attributes must be present. 

Also, unlike an `<a>` element without `href`, the absence of these attributes does not appear to be defined as a special placeholder state.

For that reason, I think it would be reasonable for these sibling rules to behave differently in this case.

**Comment by tooth-is-silver** [comment]
I think these rules should only check whether the title attribute is present, without determining whether the `<a>` or `<img>` element is functional.
I’m not sure whether checking if an `<a>` functions as intended belongs in a rule that checks for the presence of a `title` attribute.
From that perspective, I would prefer to keep checking both elements consistently.

**Comment by lumirlumir** [comment]
I understand that using the `src` attribute in an `img` tag is generally the right approach and is recommended by the HTML specification guidelines. That said, using an `img` tag without a `src` attribute is syntactically valid, and therefore allowed in both Markdown and HTML.

For example:

```html
<img id="1" />
```

<img id="1" />

I wrote `<img id="1" />` above, but it isn’t rendered. However, the tag is still present when inspected using the browser’s developer tools.

Therefore, for `require-image-title`, I think the rule should also exclude `img` tags without a `src`, `srcset`, or `alt` attribute, as well as those with `alt=""` or `aria-hidden="true"`, since they generally aren’t rendered or are intended not to be rendered in the UI.

So, to clarify:

- An `a` tag without an `href` attribute and with `aria-hidden="true"` should be excluded from the rule’s default report.
- An `img` tag without a `src`, `srcset`, or `alt` attribute, as well as one with `alt=""` or `aria-hidden="true"`, should be excluded from the rule’s default report.

A PR for each of these two cases would be welcome, and keeping them separate would be great.

**Comment by lumirlumir** [comment]
@electrohyun If it’s alright, could you create a separate issue for `require-image-title` to make tracking clearer? Thanks!

**Comment by electrohyun** [comment]
@lumirlumir Sure, no problem!
